public/app/features/logs/components/panel/LogLineDetailsDisplayedFields.tsx

```diff
@@ -54,38 +65,79 @@ export const LogLineDetailsDisplayedFields = () => {
 interface DraggableDisplayedFieldProps {
   field: string;
   index: number;
+  moveField: (srcIndex: number, destIndex: number) => void;
 }
 
-const DraggableDisplayedField = ({ field, index }: DraggableDisplayedFieldProps) => {
-  const { onClickHideField } = useLogListContext();
-  const styles = useStyles2(getStyles);
+const DraggableDisplayedField = ({ field, index, moveField }: DraggableDisplayedFieldProps) => {
   return (
     <Draggable draggableId={field} index={index}>
       {(provided) => (
-        <div ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
-          <Card noMargin className={styles.fieldCard}>
-            <div>
-              {field === LOG_LINE_BODY_FIELD_NAME ? t('logs.log-line-details.log-line-field', 'Log line') : field}
-            </div>
-            {onClickHideField && (
+        <DisplayedField key={field} field={field} index={index} moveField={moveField} provided={provided} />
+      )}
+    </Draggable>
+  );
+};
+
+const DisplayedField = ({
+  field,
+  index,
+  moveField,
+  provided,
+}: DraggableDisplayedFieldProps & { provided: DraggableProvided }) => {
+  const { detailsMode, displayedFields, onClickHideField } = useLogListContext();
+  const styles = useStyles2(getStyles, detailsMode);
+  const nextIndex = index === displayedFields.length - 1 ? 0 : index + 1;
+  const prevIndex = index === 0 ? displayedFields.length - 1 : index - 1;
+  return (
+    <div ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
+      <Card noMargin className={styles.fieldCard}>
+        <div className={styles.fieldWrapper}>
+          <div className={styles.field}>
+            {field === LOG_LINE_BODY_FIELD_NAME ? t('logs.log-line-details.log-line-field', 'Log line') : field}
+          </div>
+          {displayedFields.length > 1 && (
+            <>
+              <IconButton
+                name="arrow-down"
+                onClick={() => moveField(index, nextIndex)}
+                tooltip={t('logs.log-line-details.move-displayed-field-down', 'Move down')}
+              />
               <IconButton
-                name="times"
-                onClick={() => onClickHideField(field)}
-                tooltip={t('logs.log-line-details.remove-displayed-field', 'Remove field')}
+                name="arrow-up"
+                onClick={() => moveField(index, prevIndex)}
+                tooltip={t('logs.log-line-details.move-displayed-field-up', 'Move up')}
```

Comment:
Nit: I know that it would be harder to sort things left to right via drag&drop but this always confuses me when control elements in the UI flip an axis.